local.go

```diff
@@ -83,40 +82,42 @@ func Local(f Function, initX []float64, settings *Settings, method Method) (*Res
 	}
 
 	// actually perform the minimization
-	status, err := minimize(settings, location, method, funcInfo, stats, funcs, optLoc, startTime)
+	status, err := minimize(settings, &loc, method, funcInfo, stats, funcs, startTime)
+
+	var optLoc Location
+	copyLocation(&optLoc, loc)
 
 	// cleanup at exit
 	if settings.Recorder != nil && err == nil {
-		err = settings.Recorder.Record(*optLoc, NoEvaluation, PostIteration, stats)
+		err = settings.Recorder.Record(optLoc, NoEvaluation, PostIteration, stats)
 	}
 	stats.Runtime = time.Since(startTime)
 	return &Result{
 		Stats:    *stats,
-		Location: *optLoc,
+		Location: optLoc,
```

Comment:
This should probably promoted to a pointer too. We haven't quite answered how we're going to deal with constraints, but if they go in Location the struct will become too big to just copy everywhere. Making it a pointer seems for the best.